daemon/cluster/executor/container/controller.go

```diff
@@ -564,15 +568,9 @@ func (r *controller) matchevent(event events.Message) bool {
 	if event.Type != events.ContainerEventType {
 		return false
 	}
-
-	// TODO(stevvooe): Filter based on ID matching, in addition to name.
-
-	// Make sure the events are for this container.
-	if event.Actor.Attributes["name"] != r.adapter.container.name() {
-		return false
-	}
-
-	return true
+	r.idset.Wait()
```

Comment:
stated it on my last comment. one solution is to access state of the container inside the `matchevent`. but that would require us passing `ctx` to the`matchevent` no ?